tests/optimum_cli_export.rs

```diff
@@ -0,0 +1,120 @@
+#![cfg(feature = "online")]
+#![cfg(feature = "optimum-cli")]
+//! Test the use of the ``optimum-cli`` to pull models from the Hugging Face Hub,
+//! and generate embeddings successfully with the pulled model.
+//!
+//! Generated models from optimum can have different output types - `last_hidden_state`
+//! may not be the default output. This test is to ensure that the correct output key
+//! is used when generating embeddings.
+
+use std::{path::PathBuf, process};
+
+use fastembed::{TextEmbedding, TokenizerFiles, UserDefinedEmbeddingModel, DEFAULT_CACHE_DIR};
+
+/// Check if the ``optimum-cli`` is available.
+fn has_optimum_cli() -> bool {
+    process::Command::new("optimum-cli")
+        .arg("--help")
+        .output()
+        .is_ok()
+}
+
+/// Pull a model from the Hugging Face Hub using ``optimum-cli``.
+///
+/// This function assumes you have already checked if the ``optimum-cli`` is available.
+/// The return error will not distinguish between a missing ``optimum-cli`` and a failed download.
+fn pull_model(model_name: &str, output: &PathBuf) -> anyhow::Result<TextEmbedding> {
+    eprintln!("Pulling {model_name} from the Hugging Face Hub...");
+    process::Command::new("optimum-cli")
+        .args(&[
+            "export",
+            "onnx",
+            "--model",
+            model_name,
+            output
+                .as_os_str()
+                .to_str()
+                .expect("Failed to convert path to string"),
+        ])
+        .output()
+        .map_err(|e| anyhow::anyhow!("Failed to pull model: {}", e))?;
+
+    load_model(output)
+}
+
+/// Load bytes from a file, with a nicer error message.
+fn load_bytes_from_file(path: &PathBuf) -> anyhow::Result<Vec<u8>> {
+    std::fs::read(path).map_err(|e| anyhow::anyhow!("Failed to read file at {:?}: {}", path, e))
+}
+
+/// Load a model from a local directory.
+fn load_model(output: &PathBuf) -> anyhow::Result<TextEmbedding> {
+    let model = UserDefinedEmbeddingModel {
+        onnx_file: load_bytes_from_file(&output.join("model.onnx"))?,
+        tokenizer_files: TokenizerFiles {
+            tokenizer_file: load_bytes_from_file(&output.join("tokenizer.json"))?,
+            config_file: load_bytes_from_file(&output.join("config.json"))?,
+            special_tokens_map_file: load_bytes_from_file(&output.join("special_tokens_map.json"))?,
+            tokenizer_config_file: load_bytes_from_file(&output.join("tokenizer_config.json"))?,
+        },
```

Comment:
This might become
```suggestion
        onnx_file: load_bytes_from_file(&output.join("model.onnx"))?,
        tokenizer_file: load_bytes_from_file(&output.join("tokenizer.json"))?
```
once #101 is merged and this PR is rebased.